group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetMetadataManager.java

```diff
@@ -536,6 +538,27 @@ public OffsetFetchResponseData.OffsetFetchResponseGroup fetchAllOffsets(
             .setTopics(topicResponses);
     }
 
+    public void cleanupOffsetMetadata(List<Record> records, long offsetsRetentionMs) {
+        // TODO: get only committed offset groups?
+        offsetsByGroup.forEach((groupId, offsetsByTopic) -> {
+            try {
+                Group group = groupMetadataManager.group(groupId);
+                Map<TopicPartition, OffsetAndMetadata> expiredOffsets = group.expiredOffsets(offsetsRetentionMs, offsetsByTopic);
```

Comment:
Is there a reason why we need to pass `offsetsByTopic` to the group? Would it be possible to just ask the group is offsets could be expired and then expire them if it is possible. This would follow the pattern that we have established in @dongnuo123's PR.